hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/RpcConstants.java

```diff
@@ -37,7 +37,9 @@ private RpcConstants() {
   
   
   public static final int INVALID_RETRY_COUNT = -1;
-  
+  // Special state ID value to indicate client request header has routerFederatedState set.
+  public static final long REQUEST_HEADER_NAMESPACE_STATEIDS_SET = -2L;
```

Comment:
I think I'd rather have this constant be added along with the usage rather than preemptively.

Reply:
I agree. Removed.